File: child-dir/src/main/java/codereview/simpleorder/order/domain/OrderItem.java
package codereview.simpleorder.order.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@ToString(exclude = "order")
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_item_id")
    private Long id;

    @JoinColumn(name = "order_id")
    @ManyToOne(fetch = FetchType.LAZY)
    private Order order;

    @Column(nullable = false, length = 100)
    private Long itemId;

    @Column(nullable = false)
    private String itemName;

    @Column(nullable = false)
    private String size;

    @Column(nullable = false)
    private int orderItemPrice;

    @Column(nullable = false)
    private int orderItemQuantity;

    public OrderItem(Long itemId, String itemName, String size, int orderItemPrice, int orderItemQuantity) {
        this.itemId = itemId;
        this.itemName = itemName;
        this.size = size;
        this.orderItemPrice = orderItemPrice;
        this.orderItemQuantity = orderItemQuantity;
    }

    public void mapOrder(Order order) {
        this.order = order;
    }

    public int orderItemAmount() {
        return orderItemPrice * orderItemQuantity;
    }
}

File: order/src/main/kotlin/io/philo/shop/domain/OrderItem.kt
package io.philo.shop.domain

import jakarta.persistence.*
import lombok.AccessLevel
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.ToString

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@ToString(exclude = ["order"])
class OrderItem(
    @field:Column(nullable = false, length = 100) val itemId: Long,

    @field:Column(nullable = false) val itemName: String,

    @field:Column(nullable = false) val size: String,

    @field:Column(nullable = false) val orderItemPrice: Int, // 실제 주문 가격

    @field:Column(nullable = false) val orderedQuantity: Int // 주문 수량
) {

    constructor() : this(0L, "", "", 0, 0)

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_item_id")
    val id: Long? = null

    @JoinColumn(name = "order_id")
    @ManyToOne(fetch = FetchType.LAZY)
    var order: Order? = null

    fun mapOrder(order: Order?) {
        this.order = order
    }

    fun orderItemAmount(): Int {
        return orderItemPrice * orderedQuantity
    }
}